File: lab10/Bank Account Command Pattern/BankApp/src/Executors/CommandExecutor.java
package Executors;

import Command.Command;

import java.util.ArrayList;
import java.util.List;

public class CommandExecutor {
    private List<Command> commands = new ArrayList<>();

    public void executeCommand(Command command) {
        commands.add(command);
        command.execute();
    }

    public void commandHistory() {
        System.out.println("\nCommand History");
        for (Command cmd : commands) {
            cmd.signature();
        }
    }
}

File: Lab_10/BankApp/Executor/CommandExecutor.cs
using BankApp.Commands;

namespace BankApp.Executor;

public class CommandExecutor
{
    private List<ICommand> _commands = new();

    public void ExecuteCommand(ICommand command)
    {
        _commands.Add(command);
        command.Execute();
    }

    public void CommandHistory()
    {
        Console.WriteLine("\nCommand History");
        foreach (ICommand cmd in _commands)
        {
            cmd.Signature();
        }
    }
}